kotlinx-coroutines-debug/build.gradle.kts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2016-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+import com.github.jengelman.gradle.plugins.shadow.tasks.*
+
+apply(plugin = "com.github.johnrengelman.shadow")
+
+// apply plugin to use autocomplete for Kover DSL
+apply(plugin = "org.jetbrains.kotlinx.kover")
+
+configurations {
+    val shadowDeps by creating // shaded dependencies, not included into the resulting .pom file
+    compileOnly.get().extendsFrom(shadowDeps)
+    runtimeOnly.get().extendsFrom(shadowDeps)
+}
+
+val junit_version by properties
+val junit5_version by properties
+val byte_buddy_version by properties
+val blockhound_version by properties
+val jna_version by properties
+
+dependencies {
+    compileOnly("junit:junit:$junit_version")
+    compileOnly("org.junit.jupiter:junit-jupiter-api:$junit5_version")
+    testImplementation("org.junit.jupiter:junit-jupiter-engine:$junit5_version")
+    testImplementation("org.junit.platform:junit-platform-testkit:1.7.0")
+    add("shadowDeps", "net.bytebuddy:byte-buddy:$byte_buddy_version")
+    add("shadowDeps", "net.bytebuddy:byte-buddy-agent:$byte_buddy_version")
+    compileOnly("io.projectreactor.tools:blockhound:$blockhound_version")
+    testImplementation("io.projectreactor.tools:blockhound:$blockhound_version")
+    testImplementation("com.google.code.gson:gson:2.8.6")
+    api("net.java.dev.jna:jna:$jna_version")
+    api("net.java.dev.jna:jna-platform:$jna_version")
+}
+
+java {
+    /* This is needed to be able to run JUnit5 tests. Otherwise, Gradle complains that it can't find the
+    JVM1.6-compatible version of the `junit-jupiter-api` artifact. */
+    disableAutoTargetJvm()
+}
+
+// This is required for BlockHound tests to work, see https://github.com/Kotlin/kotlinx.coroutines/issues/3701
+tasks.withType<Test>().configureEach {
+    if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_13)) {
+        jvmArgs("-XX:+AllowRedefinitionToAddDeleteMethods")
+    }
+}
+
+val jar by tasks.getting(Jar::class) {
```

Comment:
Thank you, the Archangel, but how do I make it work in this part?
```kotlin
    if (outgoing.artifacts.buildDependencies.getDependencies(null).contains(jar)) {
        outgoing.artifacts.removeIf { it.buildDependencies.getDependencies(null).contains(jar) }
```
I feel that, because of this rule, I need the exact value of the task in any case whenever I access any configuration.

Reply:
For now, I'll suggest to use `.get()` in the `configuration.all` block, so that it restrict the "issue" to the place where we struggle to find a solution while places that could already be solved are solved. E.g. tasks that depends on `jar` or `shadowJar` tasks will already stop eagerly configure them, which is conceptually already a win.

That said, I think there might be a way to write that `configuration.all` block in a lazier way that would not require a an early `get`, but I'll have to think. I'll comment here if I found a way.

Reply:
🙏